ibis/tests/expr/test_table.py

```diff
@@ -1064,41 +1189,43 @@ def test_filter_join():
     repr(filtered)
 
 
-def test_inner_join_overlapping_column_names():
-    t1 = ibis.table([("foo", "string"), ("bar", "string"), ("value1", "double")])
-    t2 = ibis.table([("foo", "string"), ("bar", "string"), ("value2", "double")])
-
-    joined = t1.join(t2, "foo")
-    expected = t1.join(t2, t1.foo == t2.foo)
-    assert_equal(joined, expected)
-    assert joined.columns == ["foo", "bar", "value1", "bar_right", "value2"]
-
-    joined = t1.join(t2, ["foo", "bar"])
-    expected = t1.join(t2, [t1.foo == t2.foo, t1.bar == t2.bar])
-    assert_equal(joined, expected)
-    assert joined.columns == ["foo", "bar", "value1", "value2"]
-
-    # Equality predicates don't have same name, need to rename
-    joined = t1.join(t2, t1.foo == t2.bar)
-    assert joined.columns == [
-        "foo",
-        "bar",
-        "value1",
-        "foo_right",
-        "bar_right",
-        "value2",
-    ]
-
-    # Not all predicates are equality, still need to rename
-    joined = t1.join(t2, ["foo", t1.value1 < t2.value2])
-    assert joined.columns == [
-        "foo",
-        "bar",
-        "value1",
-        "foo_right",
-        "bar_right",
-        "value2",
-    ]
+# TODO(kszucs): the inner join convenience to don't duplicate the equivalent
+# columns from the right table is not implemented yet
```

Comment:
Ah, this answers my question above!

Reply:
I would prefer to defer it to a follow-up to unblock the backend PRs.